crates/usvg/src/writer.rs

```diff
@@ -945,6 +946,18 @@ fn has_xlink(tree: &Tree) -> bool {
                         return true;
                     }
                 }
+
+                if let Some(ref mask) = g.mask {
+                    if has_xlink(&mask.root) {
+                        return true;
+                    }
+
+                    if let Some(ref sub_mask) = mask.mask {
+                        if has_xlink(&sub_mask.root) {
+                            return true;
+                        }
+                    }
+                }
```

Comment:
I guess it would be nice to reuse the `node.subroots` method here, but it looks like this one just allows me to pass a function to make changes to the node itself instead of giving me a way of iterating over the nodes itself and return a boolean from that.

Reply:
Yeah... I haven't figured out a better API.